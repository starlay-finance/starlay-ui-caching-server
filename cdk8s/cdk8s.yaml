language: typescript
app: node gen/main.js
imports:
  - k8s
