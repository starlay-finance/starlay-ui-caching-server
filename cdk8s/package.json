{
  "name": "cdk8s",
  "version": "1.0.0",
  "main": "main.js",
  "types": "main.ts",
  "license": "Apache-2.0",
  "private": true,
  "scripts": {
    "import:dev": "TARGET=dev cdk8s import",
    "synth:dev": "TARGET=dev cdk8s synth",
    "synth:prod": "TARGET=prod cdk8s synth",
    "compile": "tsc",
    "watch": "tsc -w",
    "test": "TARGET=dev jest",
    "build": "npm run compile && npm run test && npm run synth",
    "upgrade": "npm i cdk8s@latest cdk8s-cli@latest",
    "upgrade:next": "npm i cdk8s@next cdk8s-cli@next"
  },
  "dependencies": {
    "cdk8s": "^1.0.0-beta.30",
    "cdk8s-plus-17": "^1.0.0-beta.57",
    "constructs": "^3.3.207"
  },
  "devDependencies": {
    "@types/jest": "^27.4.0",
    "@types/node": "^17.0.14",
    "cdk8s-cli": "^1.0.89",
    "jest": "^27.4.7",
    "ts-jest": "^27.1.3",
    "typescript": "^4.5.5"
  }
}
