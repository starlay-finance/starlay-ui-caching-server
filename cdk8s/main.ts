import { newIngress } from './lib/ingress';
import { Construct } from 'constructs';
import { App, Chart, ChartProps } from 'cdk8s';
import { KubeNamespace } from 'cdk8s-plus-17/lib/imports/k8s';
import { newApiDeployment, newApiService } from './lib/api';
import { newBackends } from './lib/backend';
import { newRedisDeployment, newRedisService } from './lib/redis';
import { newCert, newCertIssuer } from './lib/cert';
import { Environments, valueOf } from './lib/common/env';
import { NAMESPACE } from './lib/common/const';

export class CacheBackendChart extends Chart {
  constructor(scope: Construct, id: string, props: ChartProps = {}) {
    super(scope, id, props);
    new KubeNamespace(this, 'namespace', {
      metadata: { name: NAMESPACE },
    });
    newApiService(this);
    newApiDeployment(this, target);
    newBackends(this, target);
    newRedisService(this);
    newRedisDeployment(this);
  }
}

export class CertChart extends Chart {
  constructor(scope: Construct, id: string, props: ChartProps = {}) {
    super(scope, id, props);
    newCertIssuer(this, target);
    newCert(this, target);
  }
}
export class IngressChart extends Chart {
  constructor(scope: Construct, id: string, props: ChartProps = {}) {
    super(scope, id, props);
    newIngress(this, target);
  }
}

const app = new App();
const target: Environments = process.env.TARGET as Environments;
if (!target || !valueOf(target)) throw new Error('Invalid target environment');

new CacheBackendChart(app, `backend`);
new CertChart(app, 'cert');
new IngressChart(app, 'ingress');
app.synth();
