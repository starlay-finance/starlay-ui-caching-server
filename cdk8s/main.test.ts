import { CacheBackendChart } from './main';
import { Testing } from 'cdk8s';

describe('Placeholder', () => {
  test('Empty', () => {
    const app = Testing.app();
    const chart = new CacheBackendChart(app, 'test-chart');
    const results = Testing.synth(chart);
    expect(results).toMatchSnapshot();
  });
});
