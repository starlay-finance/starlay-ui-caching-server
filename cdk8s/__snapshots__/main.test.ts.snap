// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder Empty 1`] = `
Array [
  Object {
    "apiVersion": "v1",
    "kind": "Namespace",
    "metadata": Object {
      "name": "cache-dev",
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "labels": Object {
        "component": "api",
      },
      "name": "api",
      "namespace": "cache-dev",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "name": "http",
          "port": 80,
          "protocol": "TCP",
          "targetPort": 3000,
        },
      ],
      "selector": Object {
        "component": "api",
      },
      "type": "NodePort",
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "component": "api",
      },
      "name": "api",
      "namespace": "cache-dev",
    },
    "spec": Object {
      "progressDeadlineSeconds": 180,
      "replicas": 2,
      "selector": Object {
        "matchLabels": Object {
          "component": "api",
        },
      },
      "strategy": Object {
        "rollingUpdate": Object {
          "maxSurge": 1,
          "maxUnavailable": 0,
        },
        "type": "RollingUpdate",
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "component": "api",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "command": Array [
                "npm",
                "run",
                "prod",
              ],
              "env": Array [
                Object {
                  "name": "REDIS_HOST",
                  "value": "redis",
                },
                Object {
                  "name": "RPC_URL",
                  "value": "https://rpc.shibuya.astar.network:8545",
                },
                Object {
                  "name": "CHAIN_ID",
                  "value": "81",
                },
              ],
              "image": "gcr.io/vein-dev-339807/vein-dev:latest",
              "imagePullPolicy": "Always",
              "name": "main",
              "ports": Array [
                Object {
                  "containerPort": 3000,
                  "name": "http",
                  "protocol": "TCP",
                },
              ],
              "readinessProbe": Object {
                "httpGet": Object {
                  "path": "/.well-known/apollo/server-health",
                  "port": 3000,
                },
                "initialDelaySeconds": 10,
                "periodSeconds": 10,
              },
            },
          ],
        },
      },
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "component": "protocol-data-loader",
      },
      "name": "protocol-data-loader",
      "namespace": "cache-dev",
    },
    "spec": Object {
      "progressDeadlineSeconds": 180,
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "component": "protocol-data-loader",
        },
      },
      "strategy": Object {
        "rollingUpdate": Object {
          "maxSurge": 1,
          "maxUnavailable": 0,
        },
        "type": "RollingUpdate",
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "component": "protocol-data-loader",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "command": Array [
                "npm",
                "run",
                "job:update-general-reserves-data",
              ],
              "env": Array [
                Object {
                  "name": "REDIS_HOST",
                  "value": "redis",
                },
                Object {
                  "name": "RPC_URL",
                  "value": "https://rpc.shibuya.astar.network:8545",
                },
                Object {
                  "name": "CHAIN_ID",
                  "value": "81",
                },
              ],
              "image": "gcr.io/vein-dev-339807/vein-dev:latest",
              "imagePullPolicy": "Always",
              "name": "main",
              "ports": Array [
                Object {
                  "containerPort": 3000,
                  "name": "http",
                  "protocol": "TCP",
                },
              ],
            },
          ],
        },
      },
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "component": "reserve-incentives",
      },
      "name": "reserve-incentives",
      "namespace": "cache-dev",
    },
    "spec": Object {
      "progressDeadlineSeconds": 180,
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "component": "reserve-incentives",
        },
      },
      "strategy": Object {
        "rollingUpdate": Object {
          "maxSurge": 1,
          "maxUnavailable": 0,
        },
        "type": "RollingUpdate",
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "component": "reserve-incentives",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "command": Array [
                "npm",
                "run",
                "job:update-reserve-incentives-data",
              ],
              "env": Array [
                Object {
                  "name": "REDIS_HOST",
                  "value": "redis",
                },
                Object {
                  "name": "RPC_URL",
                  "value": "https://rpc.shibuya.astar.network:8545",
                },
                Object {
                  "name": "CHAIN_ID",
                  "value": "81",
                },
              ],
              "image": "gcr.io/vein-dev-339807/vein-dev:latest",
              "imagePullPolicy": "Always",
              "name": "main",
              "ports": Array [
                Object {
                  "containerPort": 3000,
                  "name": "http",
                  "protocol": "TCP",
                },
              ],
            },
          ],
        },
      },
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "component": "user-incentives",
      },
      "name": "user-incentives",
      "namespace": "cache-dev",
    },
    "spec": Object {
      "progressDeadlineSeconds": 180,
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "component": "user-incentives",
        },
      },
      "strategy": Object {
        "rollingUpdate": Object {
          "maxSurge": 1,
          "maxUnavailable": 0,
        },
        "type": "RollingUpdate",
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "component": "user-incentives",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "command": Array [
                "npm",
                "run",
                "job:update-users-incentives-data",
              ],
              "env": Array [
                Object {
                  "name": "REDIS_HOST",
                  "value": "redis",
                },
                Object {
                  "name": "RPC_URL",
                  "value": "https://rpc.shibuya.astar.network:8545",
                },
                Object {
                  "name": "CHAIN_ID",
                  "value": "81",
                },
              ],
              "image": "gcr.io/vein-dev-339807/vein-dev:latest",
              "imagePullPolicy": "Always",
              "name": "main",
              "ports": Array [
                Object {
                  "containerPort": 3000,
                  "name": "http",
                  "protocol": "TCP",
                },
              ],
            },
          ],
        },
      },
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "component": "user-data-loader",
      },
      "name": "user-data-loader",
      "namespace": "cache-dev",
    },
    "spec": Object {
      "progressDeadlineSeconds": 180,
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "component": "user-data-loader",
        },
      },
      "strategy": Object {
        "rollingUpdate": Object {
          "maxSurge": 1,
          "maxUnavailable": 0,
        },
        "type": "RollingUpdate",
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "component": "user-data-loader",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "command": Array [
                "npm",
                "run",
                "job:update-users-data",
              ],
              "env": Array [
                Object {
                  "name": "REDIS_HOST",
                  "value": "redis",
                },
                Object {
                  "name": "RPC_URL",
                  "value": "https://rpc.shibuya.astar.network:8545",
                },
                Object {
                  "name": "CHAIN_ID",
                  "value": "81",
                },
              ],
              "image": "gcr.io/vein-dev-339807/vein-dev:latest",
              "imagePullPolicy": "Always",
              "name": "main",
              "ports": Array [
                Object {
                  "containerPort": 3000,
                  "name": "http",
                  "protocol": "TCP",
                },
              ],
            },
          ],
        },
      },
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "component": "update-block-number-loader",
      },
      "name": "update-block-number-loader",
      "namespace": "cache-dev",
    },
    "spec": Object {
      "progressDeadlineSeconds": 180,
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "component": "update-block-number-loader",
        },
      },
      "strategy": Object {
        "rollingUpdate": Object {
          "maxSurge": 1,
          "maxUnavailable": 0,
        },
        "type": "RollingUpdate",
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "component": "update-block-number-loader",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "command": Array [
                "npm",
                "run",
                "job:update-block-number",
              ],
              "env": Array [
                Object {
                  "name": "REDIS_HOST",
                  "value": "redis",
                },
                Object {
                  "name": "RPC_URL",
                  "value": "https://rpc.shibuya.astar.network:8545",
                },
                Object {
                  "name": "CHAIN_ID",
                  "value": "81",
                },
              ],
              "image": "gcr.io/vein-dev-339807/vein-dev:latest",
              "imagePullPolicy": "Always",
              "name": "main",
              "ports": Array [
                Object {
                  "containerPort": 3000,
                  "name": "http",
                  "protocol": "TCP",
                },
              ],
            },
          ],
        },
      },
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "labels": Object {
        "component": "redis",
      },
      "name": "redis",
      "namespace": "cache-dev",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "name": "redis",
          "port": 6379,
        },
      ],
      "selector": Object {
        "component": "redis",
      },
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "component": "redis",
      },
      "name": "redis",
      "namespace": "cache-dev",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "component": "redis",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "component": "redis",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "image": "redis:6-alpine",
              "imagePullPolicy": "Always",
              "name": "main",
              "ports": Array [
                Object {
                  "containerPort": 6379,
                  "name": "redis",
                  "protocol": "TCP",
                },
              ],
            },
          ],
        },
      },
    },
  },
]
`;
